apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx

```diff
@@ -13,53 +14,75 @@ import {
 import { Address } from "#/subgraph/src/scripts/last-addr";
 import { getIpfsMetadata } from "@/utils/ipfsUtils";
 import { pointSystems, proposalTypes } from "@/types";
+import useSubgraphQueryByChain from "@/hooks/useSubgraphQueryByChain";
 import { CV_SCALE_PRECISION } from "@/utils/numbers";
+import { useEffect, useMemo, useState } from "react";
+import LoadingSpinner from "@/components/LoadingSpinner";
 
 export const dynamic = "force-dynamic";
 
 export type AlloQuery = getAlloQuery["allos"][number];
 
-const { urqlClient } = initUrqlClient();
-
-export default async function Pool({
+export default function Pool({
   params: { chain, poolId, garden },
 }: {
   params: { chain: string; poolId: number; garden: string };
 }) {
-  const { data } = await queryByChain<getPoolDataQuery>(
-    urqlClient,
+  const { data, fetching } = useSubgraphQueryByChain<getPoolDataQuery>(
     chain,
     getPoolDataDocument,
     { poolId: poolId, garden: garden },
+    {},
+    {
+      topic: "pool",
+      id: poolId,
+      chainId: chain,
+    },
   );
-  const strategyObj = data?.cvstrategies?.[0] as CVStrategy;
+  const strategyObj = data?.cvstrategies?.[0] as CVStrategy | undefined;
```

Comment:
should use the correct type based in the query for example
![image](https://github.com/1Hive/gardens-v2/assets/691510/741d5890-a6f9-495f-ab80-f14f076a32cc)
![image](https://github.com/1Hive/gardens-v2/assets/691510/6cd3e4ff-fa17-411f-8ed7-224b64cca52c)

It will give type error for example that its missing 'maxCVSupply' for the use in another components.

Have the correct type avoid missing data, another way to solve the last problem its also include the missing one in the query.

Reply:
https://github.com/1Hive/gardens-v2/pull/229/commits/62f3fa611d9e6d3642dc52e515e8c95100005af3